examples/tv-casting-app/linux/main.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <controller/CHIPCommissionableNodeController.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <setup_payload/SetupPayload.h>
+#include <support/CHIPMem.h>
+
+using namespace chip;
+
+#define TV_DEVICE_TYPE 35
+
+CHIP_ERROR RunTimedEventLoop(uint32_t aDelayMilliseconds)
+{
+    System::Timer * timer = nullptr;
+    if (DeviceLayer::SystemLayer.NewTimer(timer) == CHIP_NO_ERROR)
+    {
+        timer->Start(
+            aDelayMilliseconds,
+            [](System::Layer *, void *, CHIP_ERROR err) {
+                DeviceLayer::PlatformMgr().StopEventLoopTask();
+                DeviceLayer::PlatformMgr().Shutdown();
+            },
+            nullptr);
+    }
+    else
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+    DeviceLayer::PlatformMgr().RunEventLoop();
+    return CHIP_NO_ERROR;
+}
+
+void LogOnboardingPayload(chip::SetupPayload payload)
+{
+    ChipLogProgress(Zcl, "Onboarding Payload::");
+    ChipLogProgress(Zcl, "PIN: %u", payload.setUpPINCode);
+    ChipLogProgress(Zcl, "Discriminator: %u", payload.discriminator);
+    ChipLogProgress(Zcl, "Vendor ID: %u", payload.vendorID);
+    ChipLogProgress(Zcl, "Product ID: %u", payload.productID);
+}
+
+int main(int argc, char * argv[])
+{
+    CHIP_ERROR err        = CHIP_NO_ERROR;
+    int commissionerCount = 0, selectedCommissioner = CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES;
+    chip::Controller::CommissionableNodeController mCommissionableNodeController;
+    Mdns::DiscoveryFilter filter(Mdns::DiscoveryFilterType::kDeviceType, TV_DEVICE_TYPE);
+    chip::SetupPayload payload;
+    SuccessOrExit(chip::Platform::MemoryInit());
+    SuccessOrExit(chip::DeviceLayer::PlatformMgr().InitChipStack());
+
+    // Discover commissioner TVs
+    SuccessOrExit(mCommissionableNodeController.DiscoverCommissioners(filter));
+    SuccessOrExit(RunTimedEventLoop(2 * 1000));
+
+    // Display discovered commissioner TVs
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        const Mdns::DiscoveredNodeData * commissioner = mCommissionableNodeController.GetDiscoveredCommissioner(i);
```

Comment:
Sure seems like we should just have a way to enumerate discovered commissioners on the controller, instead of looping over indices in a way that is totally undocumented (there is zero documentation for `CommissionableNodeController::GetDiscoveredCommissioner`) and so on.  Please at the very least add clear documentation about proper usage of `GetDiscoveredCommissioner` and what a null return from it does and does not mean and so on.

Reply:
Added documentation to GetDiscoveredCommissioner